src/test/ui/deref-suggestion.stderr

```diff
@@ -0,0 +1,59 @@
+error[E0308]: mismatched types
+  --> $DIR/deref-suggestion.rs:18:9
+   |
+18 |     foo(s);
+   |         ^ expected struct `std::string::String`, found reference
+   |
+   = note: expected type `std::string::String`
+              found type `&std::string::String`
+   = help: here are some functions which might fulfill your needs:
+           - .escape_debug()
+           - .escape_default()
+           - .escape_unicode()
+           - .to_lowercase()
+           - .to_uppercase()
```

Comment:
Unrelated, but we really have to do something here. These suggestions are pretty silly.